src/Solvers/ErrorMethods/LSgradient.jl

```diff
@@ -0,0 +1,41 @@
+"""
+    LSgradient <: SolverError
+
+A `SolverError` structure for computing the least-squares gradient,
+    ``\\nabla f(x) = A' (A x - b)``
+
+# Fields
+- None
+"""
+struct LSgradient <: SolverError end
+
+"""
+    LSgradientRecipe <: SolverErrorRecipe
+A `SolverErrorRecipe` structure for computing the gradient of least-squares objective.
+
+# Fields
+- `gradient::AbstractVector`, `A'r`.
```

Comment:
I am not sure what `A'r` means. I am assuming it is the residual of the normal equations, but neither `A` nor `r` are defined at this point.